rad/prelude/io-utils.rad

```
@@ -45,7 +46,32 @@
   (fn [key]
     (unlines (process-with-stdout! "git" ["config" "--get" key] ""))))
 
+(def get-git-username!
+  "Get the user name stored in git config."
+  (fn []
+    (match (get-git-config! "user.name")
+      ""    (do
+                (put-str! (string-append
+                            "git config user.name is not set. "
+                            "Set the user name and rerun you command.")))
+      'name  name)))
+
+(def get-git-commit-data!
+  "Get data from a `commit` via `show` specified by `format`"
+  (fn [format commit]
+    (unlines
+      (process-with-stdout! "git" ["show" "-s" (string-append "--format=" format) commit] ""))))
```

This should probably throw an exception if the process exits with an error.